xmovie/core.py

```diff
@@ -358,12 +370,78 @@ def save_frames(self, odir, progress=False):
+            save_single_frame(
+                fig, timestep, odir=odir, frame_pattern=self.frame_pattern, dpi=self.dpi
             )
 
+
+
+    def save_frames_parallel(self, odir):
+        '''
+        Saves all frames in parallel using dask.map_blocks.
+
+        Parameters
+        ----------
+        odir : path
+            path to output directory
+        '''
+        import numpy as np
+        import dask.array as darray
+
+        da = self.data
+        framedim = self.framedim
+        
+        assert isinstance(da, xr.DataArray)
+        
+        # move 'framedim' to be last dimension if not already
+        if da.dims[-1] != framedim:
+            new_dims = list(da.dims)
+            new_dims.append(new_dims.pop(new_dims.index(framedim))) # Move framedim to last axis
+            da = da.transpose(..., framedim)
+
+        da = da.chunk({self.framedim : 1}) # DataArray needs to have length-1 chunks in framedim
+        chunk_dims = [ dim for dim in da.dims if dim != framedim ] # Create dimensions for each chunk
+               
+        def create_render_save_single_frame(xr_array, timestep):
+            fig, ax, pp = self.render_single_frame(timestep)
+            save_single_frame(
+                fig, timestep, odir=odir, frame_pattern=self.frame_pattern, dpi=self.dpi
+            )
+            return
+
+ 
+
+#        mapped_save_and_close_frames = darray.map_blocks(_chunk_wrapper,
+#                                                         da.data,
+#                                                         drop_axis=(0,1),  # drops all but 'framedim'
+#                                                         chunks=(1,),  # needed because chunking is changing 
+#                                                         dtype=np.float64,  # needed. _chunk_wrapper returns np.nan because None is not an option
+#                                                         framedim=framedim,  # this and remaining arguments go to _chunk_wrapper
+#                                                         dims=chunk_dims, coords=da.coords, attrs=da.attrs, # needed to reconstruct arrays
+#                                                         save_and_close_frame=create_render_save_single_frame)
+        def create_render_save_single_frame2(xr_array, total_time, framedim):
+            chunk_time = xr_array[framedim]
+            timestep = abs(total_time - chunk_time).argmin(framedim).item()
+            fig, ax, pp = self.render_single_frame(timestep)
+            save_single_frame(
+                fig, timestep, odir=odir, frame_pattern=self.frame_pattern, dpi=self.dpi
+            )
+            return chunk_time
+
+
+        mapped_save_and_close_frames = da.map_blocks(func=create_render_save_single_frame2,
+                                                     args=(da.time, framedim),
+                                                     template=xr.DataArray(np.ones(da.time.shape)).chunk(dict(time=1)),
```

Comment:
```suggestion
                                                     template=xr.ones_like(da[framedim]).chunk(dict({framedim: 1})),
```

Reply:
Thanks for catching the `time` vs `framedim` typo. However, if I use `xr.ones_like(da[framedim]).chunk(dict({framedim: 1}))` I get a weird error: `ValueError: cannot create a Dataset from a DataArray with the same name as one of its coordinates`

Which is why I changed it in the first place.

Reply:
ah ok then use your version with `[framedim]` instead of `.time`